doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+| 11    | Any change in open source should be aligned and upstream.                       |                     |
+
+## 1.2 Design Overview
+
+![Block Diagram](event-alarm-framework-blockdiag.png)
+
+### 1.2.1 Basic Approach
+The feature involves new development.
+Applications act as producers by writing to a table with the help of event notify library.
+Eventd reads new record in the table and processes it:
+It saves the entry in event history table; if the event has an action and if it is *RAISE*, record gets added to alarm table, severity counter in ALARM_STATS is increased.
+If the received event action is *CLEAR*, record in the ALARM table is removed and severity counter in ALARM_STATS of that alarm is reduced by 1.
+If eventd receives an event with action *ACKNOWLEDGE* from mgmt-framework, severity counter in ALARM_STATS is reduced by 1.
+If eventd receives an event with action *UNACKNOWLEDGE* from mgmt-framework, severity counter in ALARM_STATS is increased by 1.
+Eventd then informs logging API to format the log message and send the message to syslog.
+
+Any application like pmon can subscribe to tables like ALARM_STATS to act accordingly.
+
+### 1.2.2 Container
+A new container by name, eventd, is created to hold event consumer logic.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+The framework assigns an unique sequence number to each of the events sent by applications.
+
+In addition, the framework provides the following key management services:
+
+- Push model: Event/Alarm information to remote syslog hosts and subscribed gNMI clients
+- Pull model: Event/Alarm information from CLI, REST/gNMI interfaces
+- Ability to change severity of events, turn off a particular event
+- Ability to acknowledge an alarm
+
+## 2.2 Functional Description
+Event Management Framework allows applications to store "state" of the system for user to query through various north bound interfaces.
+
+# 3 Design
+## 3.1 Overview
+There are three players in the event framework. Producers, which raises events; a consumer to receive and process them as they are raised and a set of receivers one for each NBI type.
+
+Applications act as producers of events.
+
+Event consumer class in eventd container receives and processes the received event. 
+Event consumer manages received events, updates event history table, current alarm table, event_stats table and alarm_stats tables and invokes logging API, which constructs message and sends it over to syslog. 
+
+Operator can chose to change properties of events with the help of event profile. Default
+event profile is available at */etc/evprofile/default.json*. User can download the default event profile,
+modify and upload it back to the switch to apply it. 
+
+Through event profile, user can change severity of any event and also can enable/disable a event.
+
+Through CLI, REST or gNMI, event history table and current alarm table can be retrieved using various filters.
```

Comment:
For simplicity and readability, do we consider rephrasing the `event history table` ---> `event table` and `current alarm table` ---> `alarm table?

Reply:
Done